plugin/src/main/kotlin/org/gradle/github/dependencygraph/internal/DependencyGraphRenderer.kt

```diff
@@ -0,0 +1,14 @@
+package org.gradle.github.dependencygraph.internal
+
+import org.gradle.github.dependencygraph.internal.model.BuildLayout
+import org.gradle.github.dependencygraph.internal.model.ResolvedConfiguration
+import org.gradle.github.dependencygraph.internal.util.PluginParameters
+import java.io.File
+
+interface DependencyGraphRenderer {
+    fun outputDependencyGraph(pluginParameters: PluginParameters,
+                              buildLayout: BuildLayout,
+                              resolvedConfigurations: MutableList<ResolvedConfiguration>,
```

Comment:
Shouldn't this be an immutable list at the point it's passed to this API? Given this is an API contract, it would be nice to guarantee that this is an immutable list at the point where it is passed to this method.